docs/src/verification.md

```diff
@@ -0,0 +1,166 @@
+# Verification in tests
+
+## General Overview
+
+When comparing our `compiled_model` with the `framework_model` (e.g., `PyTorch`), we aim to verify whether the output from the `compiled_model` is sufficiently similar to the output from the `framework_model` (where required degree of similarity is configurable).
```

Comment:
Let's also mention what the framework model is. E.g. model that is run on host/cpu.